test/runtime/unixsocket.c

```diff
@@ -173,28 +175,23 @@ static void *uds_dgram_server(void *arg)
     int fd = (long) arg;
     struct sockaddr_un addr;
     socklen_t addr_len = sizeof(addr);
-    int client_fd;
     uint8_t readBuf[SMALLBUF_SIZE];
 
-    client_fd = accept(fd, (struct sockaddr *) &addr, &addr_len);
-    test_assert(client_fd >= 0);
+    test_assert(recvfrom(fd, readBuf, SMALLBUF_SIZE / 2, 0, (struct sockaddr *)&addr, &addr_len) ==
+            SMALLBUF_SIZE / 2);
     test_assert(addr_len > sizeof(addr.sun_family));
     test_assert(addr_len <= sizeof(addr));
     test_assert(addr.sun_family == AF_UNIX);
-    test_assert(!strcmp(addr.sun_path, CLIENT_SOCKET_PATH));
```

Comment:
I think this should be kept, it's a valid check on both Nanos and Linux.

Reply:
Yes, this should definitely be there. I think I had removed it in the middle of changes, before recvfrom was implemented in this new way and did not restore it.